### PR TITLE
Enhancing UI Consistency and Backend Refinements

### PR DESCRIPTION
**Pull request by hjamaly**
### Overview

This PR focuses on improving the overall color consistency across the app, addressing small UI and backend issues, and enhancing the user experience. Key updates include changes to button and selection colors, UI edits for improved aesthetics and clarity, as well as backend refactoring and feature improvements.

### UI Changes

**Color Updates:**

- AI Screen:
  - Changed button colors for better visual appeal and consistency.
image

- Provider Info 
 - Adjusted the color of packages when selected for improved clarity.
image

 
- Home Screen:
  - Updated the UI to display the slogan "Solvate" instead of the provider's name and address.
image

- Select Provider Screen:
  - Changed the color of the "Select the best provider near you" text to white for better contrast.
  
image

- Provider & Seeker Info Screens:
  - Edited and aligned design elements across both screens for a consistent look and feel.
- Dialog Updates:
  - Enhanced the dialog used for assigning requests to make it more visually appealing.
  
image

- Booking info : 
  - Enhanced color of chosen package
  
image

**Backend Changes**

- Field Refactoring:
  - Changed the address field from a String to a Location object to improve data representation and consistency.
- Filter Functionality Improvements:

- Fixed issues with provider filtering.
- Refactored the filtering logic to be managed within the ViewModel:
  - Users can now filter providers by location and other fields (e.g., ratings, languages) simultaneously.
  - Added tests to verify that the filtering functionality works seamlessly. (I encourage you to test this thoroughly.)
- Additional Notes

  - All changes aim to enhance the app's usability, maintainability, and aesthetic appeal.
  - Backend changes are backward-compatible with existing functionality but introduce better practices and improved user experiences.
### Testing Instructions

- Verify the updated colors and UI consistency across all mentioned screens.
- Test provider filtering:

  - Apply multiple filters (e.g., by location and rating) and confirm expected results.
  - Confirm the address field refactoring doesn’t break existing functionality or data processing.

### CODE REVIEW

**Review by manunam2002 — APPROVED** [review]
### Feedback

This PR makes key changes to the UI and the backend of the app, making it more refined and consistent, and correcting some logic errors, this was much needed to ensure everything is clean and works as expected, great job!
### Minor Remarks
- **ProviderInfo**: The white subtitle is not readable for bright service type colours, I would leave it as it was before.

**Review by yshdb5 — CHANGES_REQUESTED** [review]
### Overall Comment  
This PR introduces valuable updates across multiple test and main files, focusing on improved UI, functionality and readability. Key changes UI components and data structures strengthen the application’s robustness and appearance.

### Strong Points  
- **Test Coverage:** Updates to test files like `EndToEndTestCreateProfile.kt`, `ModifyProviderInformationTest.kt`, and `SeekerProfileTest.kt` enhance accuracy and ensure functionality aligns with expectations.  
- **UI Refinements:** Changes improve user experience with a more intuitive and appealing interface.  
- **Data Structure Improvements:** Switching from strings to `Location` objects for the seeker's address enhances data handling and consistency.  
- **Code Readability:** Refactoring in several files improve the code organization and readability.

### Areas for Improvement  
- **Comments:** I left some comments on topics and modifications that should be discussed before merging.

### Conclusion  
This PR effectively enhances functionality, user experience, and code quality. A few tweaks in the parts where I left comments would make it even better. Excellent work!

File: app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt [hunk 1]
```diff
@@ -148,64 +147,48 @@ fun TopSection(
...
-                        },
-                    fontSize = 16.sp,
-                    fontWeight = FontWeight.Medium,
-                    color = OnPrimary,
-                    textAlign = TextAlign.Start,
-                    modifier = Modifier.testTag("servicesScreenCurrentLocation"))
-              }
+        Row(
+            modifier = Modifier.testTag("SloganIcon"),
+            verticalAlignment = Alignment.CenterVertically,
```
[the remark this answers is not in the document]
Comment (hjamaly) [reply app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt:170]: You mean centered horizontally ?

[the remark this answers is not in the document]
Comment (yshdb5) [reply app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt:170]: Nope, vertically ;)

File: app/src/main/java/com/android/solvit/shared/ui/map/MapScreen.kt [hunk 2]
```diff
@@ -154,6 +156,10 @@ fun MapContent(
   GoogleMap(
       modifier = modifier.fillMaxSize().testTag("googleMap"),
       cameraPositionState = cameraPositionState,
+      properties =
+          MapProperties(
+              mapStyleOptions =
+                  MapStyleOptions.loadRawResourceStyle(LocalContext.current, R.raw.map_style)),
```
[the remark this answers is not in the document]
Comment (hjamaly) [reply app/src/main/java/com/android/solvit/shared/ui/map/MapScreen.kt:162]: Yes let discuss on it during next stand up !

**Comment by yshdb5** [comment]
Thank you for the thoughtful modifications!

**Review by yshdb5 — APPROVED** [review]

**Pull request merged**